src/Knp/Component/Pager/Paginator.php

```diff
@@ -80,48 +89,44 @@ public function setDefaultPaginatorOptions(array $options)
      * @throws \LogicException
      * @return \Knp\Component\Pager\Pagination\PaginationInterface
      */
-    public function paginate($target, $page = 1, $limit = 10, array $options = array())
+    public function paginate($target, $page = 1, $limit = 10, array $options = [])
     {
-        $limit = intval(abs($limit));
-        if (!$limit) {
-            throw new \LogicException("Invalid item per page number, must be a positive number");
+        $limit = (int) abs($limit);
+        if ($limit < 0) {
```

Comment:
This never triggers. You don't check for 0 and line before we cast to absolute value.